bats/tests/preferences/surface-invalid-args.bats

```diff
@@ -0,0 +1,61 @@
+load '../helpers/load'
+
+local_setup() {
+    if is_windows; then
+        skip "test not applicable on Windows"
+    fi
+}
+
+@test 'initial factory reset' {
+    factory_reset
+}
+
+can_set_vm_to_vz() {
+    if ! is_macos; then
+        false
+    else
+        version=$(/usr/bin/sw_vers -productVersion)
+        majorMinorVersion="${version%.*}"
+        majorVersion="${majorMinorVersion%.*}"
+        minorVersion="${majorMinorVersion#*.}"
+        if ((majorVersion >= 14)); then
+            true
+        elif ((majorVersion <= 12)); then
+            false
+        elif ((minorVersion >= 3)); then
+            true
+        else
+            case "$(uname -m)" in
+            x86_64) true ;;
+            *) false ;;
+            esac
+        fi
+    fi
+}
+
+@test 'mac-specific failure for unacceptable start setting' {
+    if ! is_macos; then
+        skip 'need a mac for the --experimental.virtual-machine.type setting'
+    elif can_set_vm_to_vz; then
```

Comment:
I have a slight preference to close `if` blocks if control flow cannot continue (because they either exit the function or throw an exception). So don't wrap the code that follows in an `else` block:

```suggestion
    fi
    if can_set_vm_to_vz; then
```

Reply:
I prefer `if ... elif ... fi`, because it's easier to see that at most one of the paths will be taken, and if there's an `else` exactly one will be taken. This is one area I think the golang people got wrong.

It's also fewer lines of code

Reply:
Yes, but using an `else` clause obscures the fact that when the `if` branch is taken, it will do an early return and never execute the code behind the `else` branch. So logically, if you use an `else` branch, the whole rest of the function should be included in that branch, and not just a few lines.

> It's also fewer lines of code

Just a single line. But this is not Perl-golf; we are going for clarity here.

Anyways, not a big deal, can always be refactored later.